wrangler-transform/src/main/java/io/cdap/wrangler/Precondition.java

```diff
@@ -88,4 +98,32 @@ public boolean apply(Row row) throws PreconditionException {
       }
     }
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext, Relation relation) {
+    if (!isConditionSQL) {
+      return new InvalidRelation("Condition is not specified as SQL");
+    }
+
+    Schema inputSchema = getSchema();
+
+    ExpressionConversionContext context = new ExpressionConversionContext();
+    SQLDialect dest = isPushdownEnabled() ? SQLDialect.BIGQUERYSQL : SQLDialect.SPARKSQL;
```

Comment:
Why should the destination be known here? The engine will decide what SQLDialect it wants the expression in.

Reply:
Resolving this for now as this is not relevant for ANSI SQL conditions.